server/game/core/ongoingEffect/effectImpl/LoseKeyword.ts

```diff
@@ -0,0 +1,22 @@
+import { KeywordName } from '../../Constants';
+import { OngoingEffectValueWrapper } from './OngoingEffectValueWrapper';
+import type { InPlayCard } from '../../card/baseClasses/InPlayCard';
+import { KeywordInstance } from '../../ability/KeywordInstance';
+
+export class LoseKeyword extends OngoingEffectValueWrapper<KeywordName> {
+    private blankedKeywordInstances: KeywordInstance[] = [];
+
+    public override apply(target: InPlayCard) {
+        const keywordInstancesToApplyTo = target.keywords.filter((keywordInstance) => keywordInstance.name === this.value);
```

Comment:
Based on the rulings I've read I believe that a blanking effect will only "see" effects that are currently actually active on the unit.

Reply:
Do you mind sharing the rulings? I couldn't find anything specifically related

Reply:
You linked it before: https://discord.com/channels/1265343874105081856/1285715527548796968
"If an ability causes a unit to “lose” a keyword, the unit ceases to have the specified keyword and any abilities granting it that keyword **when the “‘lose” effect is resolved** cease granting the keyword for the duration of the “lose” effect. Unless otherwise specified, the unit can regain that keyword through new effects."

Reply:
Ah that's fair, I didn't think about the implications of the highlighted part. I'm starting to think you're right, though I'll probably still start a thread in there just to confirm and to ask about the Imprisoned example. For now let's assume it works this way, good discussion!